spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/convert/DatastoreEntityConverterImpl.java

```diff
@@ -0,0 +1,130 @@
+
+import java.util.Map;
+import java.util.function.Function;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.datastore.Blob;
+import com.google.cloud.datastore.BlobValue;
+import com.google.cloud.datastore.BooleanValue;
+import com.google.cloud.datastore.DoubleValue;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.EntityValue;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyValue;
+import com.google.cloud.datastore.LatLng;
+import com.google.cloud.datastore.LatLngValue;
+import com.google.cloud.datastore.LongValue;
+import com.google.cloud.datastore.StringValue;
+import com.google.cloud.datastore.TimestampValue;
+import com.google.cloud.datastore.Value;
+import com.google.common.collect.ImmutableMap;
+
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastoreMappingContext;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastorePersistentEntity;
+import org.springframework.cloud.gcp.data.datastore.core.mapping.DatastorePersistentProperty;
+import org.springframework.data.convert.EntityInstantiator;
+import org.springframework.data.convert.EntityInstantiators;
+import org.springframework.data.mapping.PersistentPropertyAccessor;
+import org.springframework.data.mapping.PreferredConstructor;
+import org.springframework.data.mapping.PropertyHandler;
+import org.springframework.data.mapping.model.ParameterValueProvider;
+import org.springframework.data.mapping.model.PersistentEntityParameterValueProvider;
+
+/**
+ * Class for object to entity and entity to object conversions
+ *
+ * @author Dmitry Solomakha
+ *
+ * @since 1.1
+ */
+public class DatastoreEntityConverterImpl implements DatastoreEntityConverter {
+	private DatastoreMappingContext mappingContext;
+
+	private EntityInstantiators instantiators = new EntityInstantiators();
+
+	private static Map<Class<?>, Function<?, Value<?>>> valFactories;
+
+	static {
+		ImmutableMap.Builder<Class<?>, Function<?, Value<?>>> builder = new ImmutableMap.Builder<>();
+
+		builder.put(Blob.class, (Function<Blob, Value<?>>) BlobValue::of);
+		builder.put(Boolean.class, (Function<Boolean, Value<?>>) BooleanValue::of);
+		builder.put(Double.class, (Function<Double, Value<?>>) DoubleValue::of);
+		builder.put(Entity.class, (Function<Entity, Value<?>>) EntityValue::of);
+		builder.put(Key.class, (Function<Key, Value<?>>) KeyValue::of);
+		builder.put(LatLng.class, (Function<LatLng, Value<?>>) LatLngValue::of);
+		builder.put(Long.class, (Function<Long, Value<?>>) LongValue::of);
+		builder.put(String.class, (Function<String, Value<?>>) StringValue::of);
+		builder.put(Timestamp.class, (Function<Timestamp, Value<?>>) TimestampValue::of);
+
+		valFactories = builder.build();
+	}
+
+	DatastoreEntityConverterImpl(DatastoreMappingContext mappingContext) {
+		this.mappingContext = mappingContext;
+	}
+
+	@Override
+	@SuppressWarnings("unchecked")
```

Comment:
Can we get rid of this?

Reply:
actually we can't - it's because of mappingContext.getPersistentEntity(...), which extends AbstractMappingContext